src/charm.py

```diff
@@ -973,27 +986,27 @@ def _get_backup_key(self) -> str:
         timestamp = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
         return f"{BACKUP_KEY_PREFIX}-{self.model.name}-{timestamp}"
 
-    def _set_kv_relation_data(
+    def _is_vault_kv_role_configured(
         self,
-        relation: Relation,
-        mount: str,
-        ca_certificate: str,
+        vault: Vault,
+        label: str,
         egress_subnets: List[str],
-    ) -> None:
-        """Set relation data for vault-kv.
-
-        Args:
-            relation: Relation
-            mount: mount name
-            ca_certificate: CA certificate
-            egress_subnets: egress subnet
-        """
-        self.vault_kv.set_mount(relation, mount)
-        vault_url = self._get_relation_api_address(relation)
-        self.vault_kv.set_ca_certificate(relation, ca_certificate)
-        self.vault_kv.set_egress_subnets(relation, egress_subnets)
-        if vault_url is not None:
-            self.vault_kv.set_vault_url(relation, vault_url)
+        role_name: str,
+        secret_id: str,
```

Comment:
Similar comment, I'd suggest being quite explicit as to whether this is the vault secret id or the juju secret id